### PR TITLE
Fix errcheck issues in server/channels/app/login_test

### PR DESCRIPTION
**Pull request by Alenoda**
#### Summary
Fix errcheck issues in server/channels/app/login_test
Also changed previous "apperr" in the file to "appErr" to align with requested naming conventions.
As the defers were made into functions I added a line above and below them, splitting up the previous "paragraph", hope this is correct.

Fixes https://github.com/mattermost/mattermost/issues/29030
Jira https://mattermost.atlassian.net/browse/MM-61442

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/login_test.go [hunk 1]
```diff
@@ -48,22 +48,33 @@ func TestCWSLogin(t *testing.T) {
 
 	t.Run("Should authenticate user when CWS login is enabled and tokens are equal", func(t *testing.T) {
 		token := model.NewToken(TokenTypeCWSAccess, "")
-		defer th.App.DeleteToken(token)
+
+		defer func() {
+			err := th.App.DeleteToken(token)
+			require.NoError(t, err)
```
Comment (hanzei) [inline server/channels/app/login_test.go:54]: There are some linter error with your changes:
```
Error: channels/app/login_test.go:54:4: calling require.NoError on *github.com/mattermost/mattermost/server/public/model.AppError, please use require.Nil instead
Error: channels/app/login_test.go:65:3: calling require.NoError on *github.com/mattermost/mattermost/server/public/model.AppError, please use require.Nil instead
Error: channels/app/login_test.go:75:4: calling require.NoError on *github.com/mattermost/mattermost/server/public/model.AppError, please use require.Nil instead
```


You need to use `require.Nil` as `err` is an `*model.AppError` and not an `error`. You can also change the variable name to `appErr` to make that clear.

Please let me know if you have any questions.

File: server/channels/app/login_test.go [hunk 2]
```diff
@@ -48,22 +48,33 @@ func TestCWSLogin(t *testing.T) {
 
 	t.Run("Should authenticate user when CWS login is enabled and tokens are equal", func(t *testing.T) {
 		token := model.NewToken(TokenTypeCWSAccess, "")
-		defer th.App.DeleteToken(token)
+
```
Comment (hanzei) [inline server/channels/app/login_test.go:51]: Please remove the empty line you added.

File: server/channels/app/login_test.go [hunk 3]
```diff
@@ -48,22 +48,33 @@ func TestCWSLogin(t *testing.T) {
...
+		err = th.App.DeleteToken(token)
+		require.NoError(t, err)
 	})
 
 	t.Run("Should not authenticate the user when CWS token was used", func(t *testing.T) {
 		token := model.NewToken(TokenTypeCWSAccess, "")
 		os.Setenv("CWS_CLOUD_TOKEN", token.Token)
 		require.NoError(t, th.App.Srv().Store().Token().Save(token))
-		defer th.App.DeleteToken(token)
+
```
Comment (hanzei) [inline server/channels/app/login_test.go:72]: same here

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR :+1:

**Comment by Alenoda** [comment]
@hanzei Hopefully fixed now

**Review by hanzei — CHANGES_REQUESTED** [review]
One more left:
`Error: channels/app/login_test.go:64:3: calling require.Nil on error, please use require.NoError instead`

File: server/channels/app/login_test.go [hunk 4]
```diff
@@ -48,22 +48,31 @@ func TestCWSLogin(t *testing.T) {
...
 		os.Setenv("CWS_CLOUD_TOKEN", token.Token)
 		user, err := th.App.AuthenticateUserForLogin(th.Context, "", th.BasicUser.Username, "", "", token.Token, false)
 		require.Nil(t, err)
 		require.NotNil(t, user)
 		require.Equal(t, th.BasicUser.Username, user.Username)
-		_, apperr := th.App.Srv().Store().Token().GetByToken(token.Token)
-		require.NoError(t, apperr)
-		th.App.DeleteToken(token)
+		_, appErr := th.App.Srv().Store().Token().GetByToken(token.Token)
+		require.NoError(t, appErr)
```
Comment (hanzei) [inline server/channels/app/login_test.go:62]: `th.App.Srv().Store().Token().GetByToken` returns an `error` please don't use `appErr` as a variable name.

File: server/channels/app/login_test.go [hunk 5]
```diff
@@ -48,22 +48,31 @@ func TestCWSLogin(t *testing.T) {
...
 		token := model.NewToken(TokenTypeCWSAccess, "")
-		defer th.App.DeleteToken(token)
+		defer func() {
+			appErr := th.App.DeleteToken(token)
+			require.Nil(t, appErr)
+		}()
+
 		os.Setenv("CWS_CLOUD_TOKEN", token.Token)
 		user, err := th.App.AuthenticateUserForLogin(th.Context, "", th.BasicUser.Username, "", "", token.Token, false)
 		require.Nil(t, err)
```
Comment (hanzei) [inline server/channels/app/login_test.go:58]: `err` is in fact a `*model.AppError`. Pleas update the variable name to `appErr`

**Review by hanzei — CHANGES_REQUESTED** [review]

File: server/channels/app/login_test.go [hunk 6]
```diff
@@ -48,22 +48,31 @@ func TestCWSLogin(t *testing.T) {
...
+		require.Nil(t, appErr)
 		require.NotNil(t, user)
 		require.Equal(t, th.BasicUser.Username, user.Username)
-		_, apperr := th.App.Srv().Store().Token().GetByToken(token.Token)
-		require.NoError(t, apperr)
-		th.App.DeleteToken(token)
+		_, err := th.App.Srv().Store().Token().GetByToken(token.Token)
+		require.NoError(t, err)
+		appErr = th.App.DeleteToken(token)
+		require.NoError(t, appErr)
```
Comment (hanzei) [inline server/channels/app/login_test.go:64]: This should be require.Nil

Comment (Alenoda) [reply server/channels/app/login_test.go:64]: Should it still be require.NoError for the above err, but update just the lower appErr to require,Nil, or update both?

Sorry for needing so many updates on this PR 😅

Comment (hanzei) [reply server/channels/app/login_test.go:64]: You only need to update this line to use `require.Nil(t, appErr)`

No problem. Keep asking questions!

**Review by hanzei — APPROVED** [review]
Nice work :+1:

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @Alenoda!

**Comment by devinbinnie** [comment]
@hanzei Assigned to you to merge, not sure if we need QA or not.

**Pull request merged**